app/concerns/gkr_keyword_recommendations.rb

```diff
@@ -2,14 +2,40 @@
 module GKRKeywordRecommendations
   extend ActiveSupport::Concern
 
-  def fetch_keyword_recommendations
+  def fetch_keyword_recommendations(user,provider)
     abstract = get_resource.draft.fetch('Abstract', '')
+    
     response = cmr_client.fetch_keyword_recommendations(abstract)
-    keyword_recommendations = if response.success?
-                                response.body['recommendations'].map { |rec| rec['keyword'] }
-                              else
-                                []
-                              end
-    # TODO log recommendations (and failure)
+    
+    if response.error?
+      log_gkr_request_failed(user, provider, abstract, response)
+      return {id: "",recommendations: []}
+    end
+      
+    keyword_recommendations = response.body['recommendations'].map { |rec| rec['keyword'] }
```

Comment:
It looks like the first time we are logging (successful) recommendations is on save, not when we first request and receive the recommendations. Is that intended? I thought we were going to log when we first received the recommendations as well but maybe I am mistaken.

Reply:
Do we want to log recommendations we don't save?

Reply:
We fetch the recommendations from gkr when we first build the descriptive_keywords form. If we store them at that point, the user may close the window, walk away for lunch, etc. and will never see the recommendations.   If we store them on save, we know that they have acknowledged that they saw them.

Reply:
@ryanmiller-1 - I _thought_ we wanted to log all recommendations provided, and then we can also log what is saved and then published, to be comprehensive in what we record as we aren't sure what we're sending as feedback just yet.
@jdf-eos - that makes sense.